### PR TITLE
Code Refactor.

### PR DESCRIPTION
**Pull request by amit-webkul**

### CODE REVIEW

File: packages/Webkul/Category/src/Repositories/CategoryRepository.php [hunk 1]
```diff
@@ -294,6 +299,25 @@ public function getPartial($columns = null)
...
+    public function getSearchedCategories()
+    {
+        return $this->model->scopeQuery(function ($query) {
+            return $query
+                ->select('categories.*')
+                ->leftJoin('category_translations', function ($query) {
+                    $query->on('categories.id', '=', 'category_translations.category_id')
+                        ->where('category_translations.locale', app()->getLocale());
+                })
+                ->where('category_translations.name', 'like', '%'.urldecode(request()->input('query')).'%')
```
Comment (devansh-webkul) [inline packages/Webkul/Category/src/Repositories/CategoryRepository.php:316]: Use params variable here and remove request dependency.

$params['query'] and $params['limit']

**Review by devansh-webkul — CHANGES_REQUESTED** [review]

File: packages/Webkul/Category/src/Repositories/CategoryRepository.php (hunk 1 above)
Comment (amit-webkul) [reply packages/Webkul/Category/src/Repositories/CategoryRepository.php]: Sir done

File: packages/Webkul/Category/src/Repositories/CategoryRepository.php [hunk 2]
```diff
@@ -294,6 +292,25 @@ public function getPartial($columns = null)
...
+            ->select('categories.*')
+            ->leftJoin('category_translations', function ($join) {
+                $join->on('categories.id', '=', 'category_translations.category_id')
+                    ->where('category_translations.locale', app()->getLocale());
+            })
+            ->where('category_translations.name', 'like', '%'.urldecode($params['query']).'%')
+            ->orderBy('created_at', 'desc');
+
+        return $queryBuilder->paginate($params['limit'] ?? 10);
+    }
```
Comment (devansh-webkul) [inline packages/Webkul/Category/src/Repositories/CategoryRepository.php:312]: Can you please try to use the getAll method of this repo and reduce this method.

**Review by devansh-webkul — CHANGES_REQUESTED** [review]

File: packages/Webkul/Category/src/Repositories/CategoryRepository.php (hunk 2 above)
Comment (amit-webkul) [reply packages/Webkul/Category/src/Repositories/CategoryRepository.php]: Done.

**Review by devansh-webkul — APPROVED** [review]

**Review by jitendra-webkul — APPROVED** [review]

**Pull request merged**